### PR TITLE
Add conditional generic

### PR DESCRIPTION
**Pull request by KarimEbrahemAbdelaziz**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [Carbon](https://carbon.now.sh) Requirements
- [x] Select `Dracula` Theme
- [x] Select right `programming language`
- [x] Set `largest` font size

- Commit requirements
- [x] Image should be added to `screenshots` folder
- [x] Code snippet should be added to `codes` folder
- [x] Image and code snippet have the `same name` for their files

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,9 @@
 [Instagram]: https://www.instagram.com/SwiftDailyTips
 [Telegram]: https://t.me/SwiftDailyTips
 
+### [#22 Conditional Generic](/codes/ConditionalGeneric.swift)
```
[the remark this answers is not in the document]
Comment (KarimEbrahemAbdelaziz) [reply README.md]: Oki got it 🔥 👍🏻

File: codes/ConditionalGeneric.swift [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    }
+}
+square(CorrectStruct())  // return 25
+
+// struct IncorrectStruct {
+//     var integerValue: Int {
+//         return 5
+//     }
+// }
+// square(IncorrectValue())  // compile error
```
[the remark this answers is not in the document]
Comment (KarimEbrahemAbdelaziz) [reply codes/ConditionalGeneric.swift]: Yea, my fault sorry 👍 💯

**Comment by KarimEbrahemAbdelaziz** [comment]
I already resolve all changes requested.

**Comment by emreozdil** [comment]
Hey, this pr includes 2 tips. You create a branch for each tip. You can open pr for only public private set tip. I will merge and share your tip

**Comment by KarimEbrahemAbdelaziz** [comment]
> Hey, this pr includes 2 tips. You create a branch for each tip. You can open pr for only public private set tip. I will merge and share your tip

Hey, 
I will close this pr and create it again from a branch.
And sorry for those mistakes as it's my first open source contribution 😅

**Pull request closed without merging**